taipy/common/config/_serializer/_base_serializer.py

```diff
@@ -155,12 +159,16 @@ def _pythonify(cls, val):
                         return _TemplateHandler._to_function(actual_val)
                     elif dynamic_type == "class":
                         return _TemplateHandler._to_class(actual_val)
+                    elif dynamic_type == "enum":
+                        return _TemplateHandler._to_enum(actual_val)
                     elif dynamic_type == "str":
                         return actual_val
                     else:
                         error_msg = f"Error loading toml configuration at {val}. {dynamic_type} type is not supported."
                         raise LoadingError(error_msg)
             if isinstance(val, dict):
+                if len(val) == 1 and list(val.keys())[0] in cls._registered_types.keys():
+                    return cls._registered_types[list(val.keys())[0]]._pythonify(list(val.values())[0])
```

Comment:
This `if` command handles the case where a dataclass instance is serialized into a dictionary.

For example, UserAttribute("email", "credential_mail", default="example@taipy.org") will be serialized to

```
{ "USER_ATTRIBUTE": {"name": "email", "credentials_name": "credentials_mail", "default": "example@taipy.org"}
```
The key is used to identify the type of the serialized object.